### PR TITLE
[WFLY-18691] Add documentation for adding integrity checking for an existing filesystem realm to wildfly docs

### PR DESCRIPTION
**Pull request by PrarthonaPaul**
issue link: https://issues.redhat.com/browse/WFLY-18691

### CODE REVIEW

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc [hunk 1]
```diff
@@ -282,3 +282,30 @@ $JBOSS_HOME/bin/elytron-tool.sh filesystem-realm-encrypt -i ./standalone/configu
...
 
 A `.cli` script will also be generated at the root of the filesystem realm. The script contains WildFly CLI commands that can be used to configure a `secret-key-credential-store` resource and a `filesystem-realm` resource in the Elytron subsystem that makes use of the newly encrypted realm content.
+
+== Adding Integrity Checking for an Existing Filesystem Realm
+
+Integrity checking can also be added to an existing filesystem realm using the WildFly Elytron Tool. This takes an existing filesystem realm and creates a new filesystem realm with integrity checking support. The integrity checking uses a set of private-public key pairs, where the private key is only shared with certain known and trusted parties. As a result, the filesystem is secured and someone with access to the identity files, can no longer tamper with the data.
```
Comment (fjuma) [inline docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc:288]: s/a set of private-public key pairs/a public and private key pair

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc [hunk 2]
```diff
@@ -282,3 +282,30 @@ $JBOSS_HOME/bin/elytron-tool.sh filesystem-realm-encrypt -i ./standalone/configu
...
 
 A `.cli` script will also be generated at the root of the filesystem realm. The script contains WildFly CLI commands that can be used to configure a `secret-key-credential-store` resource and a `filesystem-realm` resource in the Elytron subsystem that makes use of the newly encrypted realm content.
+
+== Adding Integrity Checking for an Existing Filesystem Realm
+
+Integrity checking can also be added to an existing filesystem realm using the WildFly Elytron Tool. This takes an existing filesystem realm and creates a new filesystem realm with integrity checking support. The integrity checking uses a set of private-public key pairs, where the private key is only shared with certain known and trusted parties. As a result, the filesystem is secured and someone with access to the identity files, can no longer tamper with the data. 
+
+The feature also creates a CLI script that can be used to add a new filesystem-realm resource to the Elytron subsystem and modify the new filesystem realm with the added integrity checking.
```
Comment (fjuma) [inline docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc:290]: can be used to add a key-store resource and a filesystem-realm resource in the Elytron subsystem that makes use of the newly signed realm content

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc [hunk 3]
```diff
@@ -282,3 +282,30 @@ $JBOSS_HOME/bin/elytron-tool.sh filesystem-realm-encrypt -i ./standalone/configu
...
+Integrity checking can also be added to an existing filesystem realm using the WildFly Elytron Tool. This takes an existing filesystem realm and creates a new filesystem realm with integrity checking support. The integrity checking uses a set of private-public key pairs, where the private key is only shared with certain known and trusted parties. As a result, the filesystem is secured and someone with access to the identity files, can no longer tamper with the data. 
+
+The feature also creates a CLI script that can be used to add a new filesystem-realm resource to the Elytron subsystem and modify the new filesystem realm with the added integrity checking. 
+
+In order to use this feature, we can make use of the `elytron-tool.sh` (or .bat for windows) script. The format for the command to add integrity checking is as follows:
+```
+$ WILDFLY_HOME/bin/elytron-tool.sh filesystem-realm-integrity --input-location <existing_path> --output-location <new_path> --keystore <ks_path> --password <ks_password> --summary
+```
+
+The required commands include:
```
Comment (fjuma) [inline docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc:297]: s/commands/options

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc [hunk 4]
```diff
@@ -282,3 +282,30 @@ $JBOSS_HOME/bin/elytron-tool.sh filesystem-realm-encrypt -i ./standalone/configu
...
+In order to use this feature, we can make use of the `elytron-tool.sh` (or .bat for windows) script. The format for the command to add integrity checking is as follows:
+```
+$ WILDFLY_HOME/bin/elytron-tool.sh filesystem-realm-integrity --input-location <existing_path> --output-location <new_path> --keystore <ks_path> --password <ks_password> --summary
+```
+
+The required commands include: 
+* input-location: This is the path for the existing filesystem realm that does not have an integrity checking. This must be the path to the directory containing the filesystem, not the path to the filesystem itself. This can also be specified using `-i`. This must be an existing path. 
+* output-location: This attribute will be the path for the new filesystem realm. This filesystem realm will include integrity checking. This can also be specified using `-o`. This path is also a path to a directory.
+* keystore: Path for the keystore that holds the key pair for integrity checking. Can also be specified using `-k`. 
+* password: This is the password for the keystore. This will be specified by the user when creating the keystore. This can also be specified using `-p`.
```
Comment (fjuma) [inline docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc:301]: To simplify things a bit I think we can remove "This will be specified by the user when creating the keystore"

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc [hunk 5]
```diff
@@ -282,3 +282,30 @@ $JBOSS_HOME/bin/elytron-tool.sh filesystem-realm-encrypt -i ./standalone/configu
...
+
+The feature also creates a CLI script that can be used to add a new filesystem-realm resource to the Elytron subsystem and modify the new filesystem realm with the added integrity checking. 
+
+In order to use this feature, we can make use of the `elytron-tool.sh` (or .bat for windows) script. The format for the command to add integrity checking is as follows:
+```
+$ WILDFLY_HOME/bin/elytron-tool.sh filesystem-realm-integrity --input-location <existing_path> --output-location <new_path> --keystore <ks_path> --password <ks_password> --summary
+```
+
+The required commands include: 
+* input-location: This is the path for the existing filesystem realm that does not have an integrity checking. This must be the path to the directory containing the filesystem, not the path to the filesystem itself. This can also be specified using `-i`. This must be an existing path.
```
Comment (fjuma) [inline docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc:298]: I think "not the path to the filesystem itself" can be removed just to simplify the wording a bit.

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc [hunk 6]
```diff
@@ -282,3 +282,30 @@ $JBOSS_HOME/bin/elytron-tool.sh filesystem-realm-encrypt -i ./standalone/configu
...
+* input-location: This is the path for the existing filesystem realm that does not have an integrity checking. This must be the path to the directory containing the filesystem, not the path to the filesystem itself. This can also be specified using `-i`. This must be an existing path. 
+* output-location: This attribute will be the path for the new filesystem realm. This filesystem realm will include integrity checking. This can also be specified using `-o`. This path is also a path to a directory.
+* keystore: Path for the keystore that holds the key pair for integrity checking. Can also be specified using `-k`. 
+* password: This is the password for the keystore. This will be specified by the user when creating the keystore. This can also be specified using `-p`. 
+
+The elytron tool also allows for the conversion to be done in bulk, as in multiple filesystems can be converted all at once. This can be done using the following command: 
+```
+$ WILDFLY_HOME/bin/elytron-tool.sh filesystem-realm-integrity --bulk-convert <descriptor> 
+```
+In this case <descriptor> is a file, which contains the instructions for bulk conversation.
```
Comment (fjuma) [inline docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc:307]: An example of a descriptor file looks like this:
```
input-location:standalone/configuration/unsigned-realm-one
output-location:standalone/configuration/signed
keystore:myKeyStore
password:myPassword
key-pair:myKeyPair
realm-name:realm-one

input-location:standalone/configuration/unsigned-realm-two
output-location:standalone/configuration/signed
keystore:myKeyStore
password:myPassword
key-pair:myKeyPair
realm-name:realm-two
```
It just consists of the same options that would be specified when converting a single realm. There is a block of options for each realm to convert.

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc (hunk 1 above)
Comment (fjuma) [reply docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc:288]: "where the private key is only shared with certain known and trusted parties. As a result, the filesystem is secured and someone with access to the identity files, can no longer tamper with the data"

I think that could be removed and reworded to something like:

The key pair will be used to create a new filesystem realm where each identity will be signed using the private key. This makes it possible to verify that identities in the realm haven't been tampered with.

Comment (PrarthonaPaul) [reply docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc]: fixed!

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc (hunk 2 above)
Comment (PrarthonaPaul) [reply docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc]: fixed!

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc (hunk 3 above)
Comment (PrarthonaPaul) [reply docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc]: fixed!

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc (hunk 4 above)
Comment (PrarthonaPaul) [reply docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc]: fixed!

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc (hunk 5 above)
Comment (PrarthonaPaul) [reply docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc]: fixed!

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc (hunk 6 above)
Comment (PrarthonaPaul) [reply docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc]: I have added this to the docs. 
Thank you!

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc [hunk 7]
```diff
@@ -282,3 +282,45 @@ $JBOSS_HOME/bin/elytron-tool.sh filesystem-realm-encrypt -i ./standalone/configu
...
+* input-location: This is the path for the existing filesystem realm that does not have an integrity checking. This must be the path to the directory containing the filesystem. This can also be specified using `-i`. This must be an existing path. 
+* output-location: This attribute will be the path for the new filesystem realm. This filesystem realm will include integrity checking. This can also be specified using `-o`. This path is also a path to a directory.
+* keystore: Path for the keystore that holds the key pair for integrity checking. Can also be specified using `-k`. 
+* password: This is the password for the keystore. This can also be specified using `-p`. 
+
+The elytron tool also allows for the conversion to be done in bulk, as in multiple filesystems can be converted all at once. This can be done using the following command: 
+```
+$ WILDFLY_HOME/bin/elytron-tool.sh filesystem-realm-integrity --bulk-convert <descriptor> 
+```
+In this case <descriptor> is a file, which contains the instructions for bulk conversation. The contents of the descriptor file can look something like this:
```
Comment (fjuma) [inline docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc:307]: `<descriptor>`

Comment (fjuma) [reply docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc:307]: s/conversation/conversion

Comment (fjuma) [reply docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc:307]: s/The contents of the descriptor file can look something like this/An example of a descriptor file is shown below. Notice that each block contains the options to use for converting a single realm.

**Review by fjuma — CHANGES_REQUESTED** [review]
Thanks very much @PrarthonaPaul! Added a couple more small comments.

Please also create a WFLY issue for this and update the title/commit/description accordingly.

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc (hunk 7 above)
Comment (PrarthonaPaul) [reply docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc]: Thanks! 
I have added these changes and created a jira.

**Review by fjuma — APPROVED** [review]
Thanks very much @PrarthonaPaul!

File: docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc [hunk 8]
```diff
@@ -282,3 +282,45 @@ $JBOSS_HOME/bin/elytron-tool.sh filesystem-realm-encrypt -i ./standalone/configu
...
 
 A `.cli` script will also be generated at the root of the filesystem realm. The script contains WildFly CLI commands that can be used to configure a `secret-key-credential-store` resource and a `filesystem-realm` resource in the Elytron subsystem that makes use of the newly encrypted realm content.
+
+== Adding Integrity Checking for an Existing Filesystem Realm
+
+Integrity checking can also be added to an existing filesystem realm using the WildFly Elytron Tool. This takes an existing filesystem realm and creates a new filesystem realm with integrity checking support. The integrity checking uses a public and private key pair. The key pair will be used to create a new filesystem realm where each identity will be signed using the private key. This makes it possible to verify that identities in the realm haven't been tampered with. 
+
+The feature also creates a CLI script that can be used to add a key-store resource and a filesystem-realm resource in the Elytron subsystem that makes use of the newly signed realm content.
```
Comment (PrarthonaPaul) [inline docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc:290]: Should I add a note here about needing to edit the cli script for domain mode?

Comment (fjuma) [reply docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc:290]: Hi @PrarthonaPaul, since there are some other Elytron Tool commands that also produce similar CLI scripts, me/Diana are going to do a quick check to see if we have notes about domain mode for those already and we'll create a follow-up issue and PR that adds some notes for all affected commands, including this one.

So we can go ahead and merge this PR. Thanks again!

Comment (PrarthonaPaul) [reply docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc:290]: Sounds good. Thanks!

**Comment by bstansberry** [comment]
Thanks @PrarthonaPaul and @fjuma
